app/services/hyrax/edit_permissions_service.rb

```diff
@@ -0,0 +1,178 @@
+    #   - unauthorized_collection_managers [Array hashes] manager ids & collection_ids [{:name, :id}]
+    def manager_permissions_to_block
+      unauthorized_managers = []
+      unauthorized_collection_managers = []
+      if object_unauthorized_collection_ids.any?
+        object_unauthorized_collection_ids.each do |id|
+          Hyrax::PermissionTemplate.find_by(source_id: id).access_grants.each do |grant|
+            if grant.access == "manage"
+              unauthorized_managers << grant.agent_id
+              unauthorized_collection_managers += Array.wrap({ name: grant.agent_id }.merge(id: id))
+            end
+          end
+        end
+      end
+      BlockedPermissions.new(unauthorized_managers, unauthorized_collection_managers)
+    end
+
+    # find all of the work's collections a user can manage
+    # @return [Array] of collection ids
+    def object_managed_collection_ids
+      @object_managed_collection_ids ||= object_member_of & managed_collection_ids
+    end
+
+    # find all of the work's collections a user cannot manage
+    # note: if the collection type doesn't include "sharing_applies_to_new_works", we don't limit access
+    # @return [Array] of collection ids with limited access
+    def object_unauthorized_collection_ids
+      @object_unauthorized_collection_ids ||= begin
+                                                limited_access = []
+                                                unauthorized_collection_ids = object_member_of - object_managed_collection_ids
+                                                if unauthorized_collection_ids.any?
+                                                  unauthorized_collection_ids.each do |id|
+                                                    # TODO: Can we instead use a SOLR query?  This seems to be somewhat expensive.  However, as this is
+                                                    # used in administration instead of user front-end displays, I'm not as concerned.
+                                                    collection = ActiveFedora::Base.find(id)
+                                                    limited_access << id if (collection.instance_of? AdminSet) || collection.share_applies_to_new_works?
+                                                  end
+                                                end
+                                                limited_access
+                                              end
+    end
+
+    # find all of the collection ids an object is a member of
+    # @return [Array] array of collection ids
+    def object_member_of
+      @object_member_of ||= begin
+                              belongs_to = []
+                              # get all of work's collection ids from the form
+                              belongs_to += @object.member_of_collection_ids
+                              belongs_to << @object.admin_set_id unless @object.admin_set_id.empty?
```

Comment:
is this `#empty?` catching a `""` string as id?

Reply:
I don't know; I suspect it's a `nil` test. I'm going to switch it to `if @object.admin_set_id.present?` as I always prefer to remove double negations (unless empty feels like I need one more hoop than if present).